hack/tools/logcheck/main.go

```diff
@@ -110,3 +119,40 @@ func isUnstructured(fName string) bool {
 
 	return false
 }
+
+// isKeysValid check if all keys in keyAndValues is string type
+func isKeysValid(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string) {
+	if len(keyValues)%2 != 0 {
+		pass.Report(analysis.Diagnostic{
+			Pos:     fun.Pos(),
+			Message: fmt.Sprintf("Invalid Number of arguments for %s", funName),
+		})
+	}
+
+	for index, arg := range keyValues {
+		if index%2 != 0 {
+			continue
+		}
+		if lit, ok := arg.(*ast.BasicLit); ok {
+			if lit.Kind != token.STRING {
+				pass.Report(analysis.Diagnostic{
+					Pos:     fun.Pos(),
+					Message: fmt.Sprintf("Invalid value type for key %v", lit.Value),
+				})
+				continue
+			}
+			isASCII := utf8string.NewString(lit.Value).IsASCII()
+			if !isASCII {
+				pass.Report(analysis.Diagnostic{
+					Pos:     fun.Pos(),
+					Message: fmt.Sprintf("Invalid value for key %v, it's must be ascii", lit.Value),
+				})
+			}
+		} else {
+			pass.Report(analysis.Diagnostic{
+				Pos:     fun.Pos(),
+				Message: "Invalid value type for key",
```

Comment:
What kind of failures we expect here? Can we give some suggestions to users about fixing this?